### PR TITLE
Fix adam docstring (L2 implementation)

### LINKED ISSUE #48793: Weight_decay in torch.Adam
## 📚 Documentation

In the current [pytorch docs for torch.Adam](https://pytorch.org/docs/stable/optim.html), the following is written:

"Implements Adam algorithm.

It has been proposed in [Adam: A Method for Stochastic Optimization](https://arxiv.org/abs/1412.6980). The implementation of the L2 penalty follows changes proposed in [Decoupled Weight Decay Regularization](https://arxiv.org/abs/1711.05101)."

This would lead me to believe that the current implementation of Adam is essentially equivalent to AdamW.  The fact that torch.AdamW exists as a separate optimizer leads me to believe that this isn't true.  Also, after looking at the source code of torch.Adam, I don't see any difference from a standard L2 penalty implementation.  So is the documentation incorrect and the "changes proposed in [Decoupled Weight Decay Regularization]" are actually absent from torch.Adam?

cc @jlin27 @mruberry @vincentqb

### PR DESCRIPTION
**Pull request by agamemnonc**
Fixes #48793. 

Corrects `torch.optim.Adam` docstring with regards to L2-penalty implementation.

### CODE REVIEW

**Comment by iramazanli** [comment]
thank you for sending this amazing PR :) 

I do agree that this issue needed to be fixed, and actually in an ongoing documentation project this issue has been addressed! That's why we cannot merge this PR currently, but we appreciate your warning us in this issue. We will be looking forward for your further contributions.
